projects/dxc-ngx-cdk/src/lib/dxc-file-input/dxc-file-input.component.ts

```diff
@@ -277,10 +300,127 @@ export class DxcFileInputComponent implements OnChanges, OnInit {
         this.service.emptyArrayFiles();
       }
       this.getPreviewsFiles(event.target.files);
+      this.processFiles(event.target.files);
       event.target.value = "";
     }
   }
 
+    /**
+   * File upload logic to send file as chunk and receive response.
+   * @param event
+   */
+    processFiles(event) {
+    this.files=event;
+    this.fileEventType = EventType.UPLOAD;
+    for(let i=0;i<event.length;i++)
+    {
+      this.uploadChunkDoc(event[i])
+    }
+  }
+
+  UploadFile(eventFiles) {
+  this.uploadChunkDoc(eventFiles);
+}
+uploadChunkDoc(file) {
+  let lastChunksize = 0;
+  this.fileDataUpload = new FileMetaData();
+  this.GUID = uuidv4();
+  this.globalActualChunkCount = 0;
+  this.globalChunkCount = file.size % 1000000 == 0 ? file.size / 1000000 : Math.floor(file.size / 1000000) + 1;
+
+  this.readFile(file, lastChunksize, this.uploadtoAPI.bind(this));
+ }
+
+ uploadtoAPI(filedata,file, lastChunksize, result) {
+  lastChunksize = lastChunksize + 1000000;
+  this.chunkResult = result;
+  if(result) {
+    //Add you logic what do you want after reading the file
+      this.uploads(filedata);
+      this.readFile(file, lastChunksize, this.uploadtoAPI.bind(this));
+
+  }
+  else if(this.globalActualChunkCount == this.globalChunkCount)
+  {
+    setTimeout(() => {
+      this.uploadcomplete(this.fileDataUpload).then(resp => { 
+        this.data[0].postresponse = resp;
+        this.fileEventType = EventType.POSTUPLOAD;
+        //let data = this.getPreview(file);
+        this.data[0].eventtype = this.fileEventType;
+        this.callbackFile.emit(this.data);
+        });
+    }, 3000);
+  }
+ }
+
+ readFile(file,lastChunksize: number, callback) {
+  let filedata = new chunkmetadata();
+  filedata.fileName = this.globalActualChunkCount + "-" + this.GUID + file.name;
+   filedata.fileSize = file.size;
+   filedata.fileType = file.type;
+   this.fileDataUpload.GUID = this.GUID;
+  var chunk = file.slice(lastChunksize,lastChunksize+1000000);
+  if(chunk.size !=0) {
+    let fileReader = new FileReader();
+    fileReader.onloadend= (result)=>{
+      // Store base64 encoded representation of file
+    filedata.fileAsBase64 = fileReader.result.toString();
+    this.globalActualChunkCount++;
+    this.fileDataUpload.fileNames.push(filedata.fileName);
+    return callback(filedata,file,lastChunksize,fileReader.result)
+    }
+    fileReader.readAsDataURL(chunk);
+  }else {
+   return callback(filedata,file,lastChunksize,false);
+  }
+ }
+ private async uploads(theFile: chunkmetadata) 
+ {
+   const response = await fetch(this.requests.uploadrequest.url, {
```

Comment:
uploadrequest should be camel case uploadRequest

Reply:
changed to camelcase.